shelljs/shelljs.d.ts

```diff
@@ -434,31 +436,44 @@ declare module "shelljs"
      * Object containing environment variables (both getter and setter). Shortcut to process.env.
      */
     export var env: { [key: string]: string };
-
-    /*
-
-    // Not yet implemented due to implementation issues (constant overloads and return types).
-    // See: https://github.com/arturadib/shelljs#execcommand--options--callback
-
-    export function exec(command: string, options: ExecOptions, callback: (code: number, output: string) => any): any;
-    export function exec(command: string, options: ExecOptions): any;
-
-    interface ExecOptions
-    {
-        silent: boolean;
-        async: boolean;
-    }
-
-    */
-
+    
     /**
      * Executes the given command synchronously.
-     * @param  {string}          command The commadn to execute.
+     * @param  {string}          command The command to execute.
      * @return {ExecReturnValue}         Returns an object containing the return code and output as string.
      */
-    export function exec(command: string): ExecReturnValue;
+    export function exec(command: string): ExecOutputReturnValue;
+    /**
+     * Executes the given command synchronously.
+     * @param  {string}          command The command to execute.
+     * @param  {ExecOptions}     options Silence and synchronous options.
+     * @return {ExecReturnValue}         Returns an object containing the return code and output as string.
+     */
+    export function exec(command: string, options: ExecOptions): ExecOutputReturnValue;
+    /**
+     * Executes the given command synchronously.
+     * @param  {string}          command  The command to execute.
+     * @param  {ExecOptions}     options  Silence and synchronous options.
+     * @param  {ExecCallback}    callback Receives code and output asynchronously.
+     */
+    export function exec(command: string, options: ExecOptions, callback: ExecCallback): child.ChildProcess;
```

Comment:
If a callback is specified, it will convert async to true, overriding what is passed in, and go into async mode, and always return a ChildProcess.  So this is correct.